src/cache-key.ts

```diff
@@ -0,0 +1,51 @@
+import type { CacheKeyBuilder } from './types'
+import { hash as ohash } from 'ohash'
+
+function sanitize(input?: string) {
+  if (!input)
+    return ''
+  return input.replace(/[^\w.-]/g, '_')
+}
+
+function join(...parts: Array<string | number>) {
+  const out: string[] = []
+  for (const p of parts) {
+    const s = sanitize(String(p))
+    if (!s)
+      continue
+    out.push(s)
+  }
+  return out.join('-')
+}
+
+/**
+ * Create a cache-key factory bound to a provider and its options.
+ * Format: `<provider>:<hash>-<body?>-<label>` (body omitted if empty).
+ * All parts are sanitized to `[\w.-]` and joined with `-`.
+ *
+ * Example:
+ * ```ts
+ * const cacheKey = createCacheKeyFactory('google', { experimental: {} })
+ * // Meta only
+ * const metaKey = cacheKey('meta.json')
+ * // Data key (family + resolve options hash)
+ * const dataKey = cacheKey('data.json', ({ hash, join }) => join('Roboto', hash({ weights: ['400'], styles: ['normal'], subsets: ['latin'] })))
+ * ```
+ */
+export function createCacheKeyFactory(providerName: string, providerOptions: unknown) {
+  const provider = sanitize(providerName)
+  const providerHash = ohash(providerOptions)
+
+  return (label: string, build?: CacheKeyBuilder) => {
```

Comment:
I think that could be enough. If it is an object, then it would be hashes automatically
```suggestion
  return (label: string, ...rest: (string | Record<string, any>)[]) => {
```

Reply:
@florian-lefebvre 
Thanks! I agree that API looks easier to use.

I’m torn between leaving users to call the cacheKey builder explicitly or wrapping the Storage interface so cache keys are built automatically.

The current approach is more transparent, but wrapping storage would let existing users get safe cache keys without any changes.